pkg/patch/patch.go

```diff
@@ -337,6 +326,34 @@ func patchWithContext(ctx context.Context, ch chan error, image, reportFile, pat
 	return err
 }
 
+// resolvePatchedTag merges explicit tag & suffix rules, returning the final patched tag.
+func resolvePatchedTag(imageRef reference.Named, explicitTag, suffix string) (string, error) {
+	// if user explicitly sets a final tag, that wins outright
+	if explicitTag != "" {
+		return explicitTag, nil
+	}
+
+	// parse out any existing tag from the image ref
+	var baseTag string
+	if tagged, ok := imageRef.(reference.Tagged); ok {
+		baseTag = tagged.Tag()
+	}
+
+	// if suffix is empty, default to "patched"
+	if suffix == "" {
```

Comment:
I dont think it will reach this case since we have a default value?

Reply:
This part is in case the user passes `--tag-suffix=""` explicitly. I am not sure why one would want to do that but good to cover all edge cases, I guess. Please let me know if I can amend this logic in any way to be more consistent with copa standards.

Reply:
got it, that makes sense!